activerecord/test/cases/adapters/mysql2/connection_test.rb

```diff
@@ -58,7 +58,7 @@ def test_successful_reconnection_after_timeout_with_verify
   def test_execute_after_disconnect
     @connection.disconnect!
 
-    error = assert_raise(ActiveRecord::StatementInvalid) do
+    error = assert_raise(ActiveRecord::ConnectionNotEstablished) do
```

Comment:
Out of curiosity, should this be considered a breaking change since a different error is being raised? While the original is still accessible via Exeception#cause, code that do rescue ActiveRecord::StatementInvalid will break.

Reply:
hm, I agree with your point that this is considered a breaking change. I'll revert this PR for now.